### PR TITLE
Issue #727 

### PR DESCRIPTION
**Pull request by DBcreator**
## Before submitting (must do checklist)

- [x] Did you read the [contribution guide](https://github.com/tinkoff-ai/etna/blob/master/CONTRIBUTING.md)?
- [x] Did you update the docs? We use Numpy format for all the methods and classes.
- [ ] Did you write any new necessary tests?
- [x] Did you update the [CHANGELOG](https://github.com/tinkoff-ai/etna/blob/master/CHANGELOG.md)?

### CODE REVIEW

**Review by Mr-Geekman — CHANGES_REQUESTED** [review]
1. Add examples.
2. Improve `plot_feature_relevance ` instead of creating a new plot function.

**Comment by DBcreator** [comment]
```python3
import matplotlib.pyplot as plt
import numpy as np
import pandas as pd

from etna.analysis import plot_feature_relevance
from etna.datasets import TSDataset
from etna.analysis import StatisticsRelevanceTable
from etna.transforms import (
    LagTransform,
    FourierTransform
)


def main():
    df = pd.read_csv("examples/data/example_dataset.csv", parse_dates=["timestamp"])
    df_wide = TSDataset.to_dataset(df)
    ts = TSDataset(df=df_wide, freq="D")
    lags = LagTransform(in_column="target", lags=list(range(10, 17)), out_column="lags")
    fourier = FourierTransform(period=360.25, order=4)
    ts.fit_transform([lags,fourier])
    
    relevance_table = StatisticsRelevanceTable()
    plot_feature_relevance(
        ts=ts,
        mode='p-values',
        relevance_table=relevance_table,
        relevance_aggregation_mode="per-segment",
        alpha = 0.05,
        top_k=20, 
    )
    plt.savefig("Per-segment p-values")
    plot_feature_relevance(
        ts=ts,
        mode='p-values',
        relevance_table=relevance_table,
        relevance_aggregation_mode="mean",
        alpha = 0.05,
        top_k=20, 
    )
    plt.savefig("Mean p-values")
    



if __name__ == "__main__":



    main()
```
Per-segment:
Per-segment p-values
Mean:
Mean p-values

File: etna/analysis/plotters.py [hunk 1]
```diff
@@ -1174,12 +1218,21 @@ def plot_feature_relevance(
...
-        ax.set_title("Feature relevance")  # type: ignore
-        ax.grid()  # type: ignore
+        if mode == "p-values":
+            relevance.sort_values(inplace=True)
+            index = relevance.index
+            pvalues, new_alpha = get_fictitious_pvalues(relevance, alpha)
+            _, ax = plt.subplots(figsize=figsize, constrained_layout=True)
+            sns.barplot(x=pvalues, y=index, orient="h", ax=ax)
+            ax.axvline(x=new_alpha)  # type: ignore
+            ax.set_title("P-values relevance:")  # type: ignore
```
Comment (Mr-Geekman) [inline etna/analysis/plotters.py:1228]: Where is `ax.grid()` like for `else` block below?

File: etna/analysis/plotters.py [hunk 2]
```diff
@@ -1132,6 +1159,8 @@ def plot_feature_relevance(
         :py:class:`~etna.analysis.feature_relevance.relevance.RelevanceTable`
     top_k:
         number of best features to plot, if None plot all the features
+    alpha:
```
Comment (Mr-Geekman) [inline etna/analysis/plotters.py:1162]: This parameter is relevant only for `StatisticsRelevanceTable`, we should emphasize it in documentation.

File: etna/analysis/plotters.py [hunk 3]
```diff
@@ -1099,13 +1102,37 @@ def plot_trend(
...
+    pvalues = coef * pvalues
+    new_alpha = coef * (alpha + eps)
+    return pvalues, new_alpha
+
+
 def plot_feature_relevance(
     ts: "TSDataset",
-    relevance_table: RelevanceTable,
+    relevance_table: Union[RelevanceTable, StatisticsRelevanceTable],
+    mode: Union[Literal["relevance"], Literal["p-values"]],
```
Comment (Mr-Geekman) [inline etna/analysis/plotters.py:1130]: Do we really need this parameter? According to the code the mode is determined by the type of `relevance_table`.

File: etna/analysis/plotters.py [hunk 4]
```diff
@@ -1099,13 +1102,37 @@ def plot_trend(
...
+    coef = 100 / sum(np.array(pvalues))
+    pvalues = coef * pvalues
+    new_alpha = coef * (alpha + eps)
+    return pvalues, new_alpha
+
+
 def plot_feature_relevance(
     ts: "TSDataset",
-    relevance_table: RelevanceTable,
+    relevance_table: Union[RelevanceTable, StatisticsRelevanceTable],
```
Comment (Mr-Geekman) [inline etna/analysis/plotters.py:1129]: We should explain in documentation that logic behind this two types of tables are different and different values are shown.

Comment (Mr-Geekman) [reply etna/analysis/plotters.py:1128]: Remain here `relevance_table: RelevanceTable`. It is a supertype for both `StatisticsRelevanceTable` and `ModelRelevanceTable`

File: etna/analysis/plotters.py [hunk 5]
```diff
@@ -1121,6 +1146,8 @@ def plot_feature_relevance(
         TSDataset with timeseries data
     relevance_table:
         method to evaluate the feature relevance
+        RelevanceTable uses relevance table with feature importance from model
```
Comment (Mr-Geekman) [inline etna/analysis/plotters.py:1149]: Here probably `ModelRelevanceTable` is meant.

According to the [rendered documentation](https://deploy-preview-795--etna-docs.netlify.app/api/etna.analysis.plotters.plot_feature_relevance.html?highlight=plot_feature_relevance#etna.analysis.plotters.plot_feature_relevance) (you can find the link in conversation of this PR), this part is rendered in a single line, it is not nice.

Read in our contribution guide about formatting documentation according to ReST markup language.

I would suggest to write it like this:
```
relevance_table:
        method to evaluate the feature relevance;

        * if :py:class:`~etna.analysis.feature_relevance.relevance.StatisticsRelevanceTable` table is used then relevances are normalized p-values

        * if :py:class:`~etna.analysis.feature_relevance.relevance.ModelRelevanceTable` table is used then relevances are importances from some model

```

File: etna/analysis/plotters.py [hunk 6]
```diff
@@ -1132,6 +1159,8 @@ def plot_feature_relevance(
         :py:class:`~etna.analysis.feature_relevance.relevance.RelevanceTable`
     top_k:
         number of best features to plot, if None plot all the features
+    alpha:
+        significance level, default alpha = 0.05, only for StatisticsRelevanceTable
```
Comment (Mr-Geekman) [inline etna/analysis/plotters.py:1163]: not StatisticsRelevanceTable but :py:class:`~etna.analysis.feature_relevance.relevance.StatisticsRelevanceTable` it will render nicely in the code

File: etna/analysis/plotters.py [hunk 7]
```diff
@@ -1159,10 +1191,18 @@ def plot_feature_relevance(
                     f"Relevances on segment: {segment} of features: {na_relevance_features} can't be calculated."
                 )
             relevance = relevance.dropna()[:top_k]
-            if normalized:
-                relevance = relevance / relevance.sum()
-            sns.barplot(x=relevance.values, y=relevance.index, orient="h", ax=ax[i])
-            ax[i].set_title(f"Feature relevance: {segment}")
+            if type(relevance_table) is StatisticsRelevanceTable:
```
Comment (Mr-Geekman) [inline etna/analysis/plotters.py:1194]: Checking type is bad, better to check `isinstance(relevance_table, StatisticsRelevanceTable)`

File: etna/analysis/plotters.py [hunk 8]
```diff
@@ -1120,7 +1145,12 @@ def plot_feature_relevance(
...
         TSDataset with timeseries data
     relevance_table:
-        method to evaluate the feature relevance
+        method to evaluate the feature relevance;
+
+        * if :py:class:`~etna.analysis.feature_relevance.relevance.StatisticsRelevanceTable` table is used then relevances are normalized p-values
+
+        * if :py:class:`~etna.analysis.feature_relevance.relevance.ModelRelevanceTable` table is used then relevances are importances from some model
+
     normalized:
```
Comment (Mr-Geekman) [inline etna/analysis/plotters.py:1154]: I think it is strange that we don't use this property if `StatisticsRelevanceTable` is used. Probably we should normalize to sum up to 1 in this case too.

Comment (Mr-Geekman) [reply etna/analysis/plotters.py:1154]: Or we should mark that these value is used only for `ModelRelevanceTable`.

File: etna/analysis/plotters.py (hunk 4 above)
Comment (Mr-Geekman) [reply etna/analysis/plotters.py:1128]: I think that we should keep this parameter but in `get_fictitious_pvalues` make normalization for sum up to 1 by default.

File: etna/analysis/plotters.py [hunk 9]
```diff
@@ -1099,13 +1100,37 @@ def plot_trend(
         ax[i].legend()
 
 
+def get_fictitious_relevances(pvalues: pd.DataFrame, alpha: float) -> Tuple[np.ndarray, float]:
+    """
+    Convert p-values into fictitious variables, with function f(x) = 1 - x.
```
Comment (Mr-Geekman) [inline etna/analysis/plotters.py:1105]: Add blank line after this line. Below we have extended description that should be separated.

File: etna/analysis/plotters.py [hunk 10]
```diff
@@ -1099,13 +1100,37 @@ def plot_trend(
...
+    Also converts alpha into fictitious variable.
+
+    Parameters
+    ----------
+    pvalues:
+        dataFrame with pvalues
+    alpha:
+        significance level, default alpha = 0.05
```
Comment (Mr-Geekman) [inline etna/analysis/plotters.py:1113]: Add blank line after this line.

File: etna/analysis/plotters.py [hunk 11]
```diff
@@ -1115,12 +1140,23 @@ def plot_feature_relevance(
 
     The most important features are at the top, the least important are at the bottom.
 
+    For StatisticsRelevanceTable also plot vertical line: fictitious significance level.
```
Comment (Mr-Geekman) [inline etna/analysis/plotters.py:1143]:
```
:py:class:`~etna.analysis.feature_relevance.relevance.StatisticsRelevanceTable`
```

File: etna/analysis/plotters.py [hunk 12]
```diff
@@ -1143,14 +1181,14 @@ def plot_feature_relevance(
...
     is_ascending = not relevance_table.greater_is_better
     features = list(set(ts.columns.get_level_values("feature")) - {"target"})
-    relevance_df = relevance_table(df=ts[:, :, "target"], df_exog=ts[:, :, features], **relevance_params).loc[segments]
-
+    relevance_df = relevance_table(df=ts[:, segments, "target"], df_exog=ts[:, segments, features], **relevance_params)
     if relevance_aggregation_mode == "per-segment":
         _, ax = prepare_axes(num_plots=len(segments), columns_num=columns_num, figsize=figsize)
         for i, segment in enumerate(segments):
+
             relevance = relevance_df.loc[segment].sort_values(ascending=is_ascending)
```
Comment (Mr-Geekman) [inline etna/analysis/plotters.py:1192]: I think like `is_ascending` isn't necessary here anymore.

File: etna/analysis/plotters.py [hunk 13]
```diff
@@ -1159,9 +1197,18 @@ def plot_feature_relevance(
...
             relevance = relevance.dropna()[:top_k]
-            if normalized:
+            if isinstance(relevance_table, StatisticsRelevanceTable):
+                relevance.sort_values(inplace=True)
+                relevance, border_value = get_fictitious_relevances(
+                    relevance,
+                    alpha,
+                )
+
+            if normalized and not isinstance(relevance_table, StatisticsRelevanceTable):
```
Comment (Mr-Geekman) [inline etna/analysis/plotters.py:1207]: Why don't we make normalization for `StatisticsRelevanceTable`?
